pkgs/dart_tooling_mcp_server/lib/src/mixins/dart_cli.dart

```diff
@@ -113,22 +165,62 @@ base mixin DartCliSupport on ToolsSupport, LoggingSupport {
           content: [
             TextContent(
               text:
-                  '$commandName failed in ${projectRoot.path}:\n$output\n\n'
-                  'Errors\n$errors',
+                  '$commandDescription failed in ${projectRoot.path}:\n'
+                  '$output\n\nErrors\n$errors',
             ),
           ],
           isError: true,
         );
       }
       if (output.isNotEmpty) {
         outputs.add(
-          TextContent(text: '$commandName in ${projectRoot.path}:\n$output'),
+          TextContent(
+            text: '$commandDescription in ${projectRoot.path}:\n$output',
+          ),
         );
       }
     }
     return CallToolResult(content: outputs);
   }
 
+  static final dartPubTool = Tool(
```

Comment:
I think we should try and make this a general "pub" tool, and it should detect whether it should use `dart pub` or `flutter pub` in each of the given roots based on project type.

Reply:
Do `dart pub` and `flutter pub` actually behave differently? Or does `flutter pub` just call into `dart pub` from Flutter tools?

Reply:
They do behave differently, `dart pub` does not work in flutter projects and `flutter pub` does not work in dart projects.

Reply:
I also just checked on a flutter project and `dart pub` works there as well.

Reply:
```
Because counter_app depends on flutter_test from sdk which doesn't exist (the
  Flutter SDK is not available), version solving failed.

Flutter users should use `flutter pub` instead of `dart pub`.
```

Reply:
(it doesn't work if you have any flutter SDK deps)

Reply:
I used `dart pub add` on a flutter project and it succeeded, but my `dart` is the Dart from the Flutter SDK so maybe that is the difference. I filed https://github.com/dart-lang/ai/issues/81 to track work and discuss the best way to determine whether a project is a dart or flutter project.